apple/internal/local_provisioning_profiles.bzl

```diff
@@ -0,0 +1,80 @@
+"""
+Bazel rules for using locally installed provisioning profiles for debug builds
+"""
+
+def _local_provisioning_profiles(repository_ctx):
+    repository_ctx.symlink(
+        "{}/Library/MobileDevice/Provisioning Profiles".format(repository_ctx.os.environ["HOME"]),
+        "profiles",
+    )
+
+    repository_ctx.file("BUILD", """
+filegroup(
+    name = "profiles",
+    srcs = glob(["profiles/*.mobileprovision"]),
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "empty",
+    srcs = [],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "fallback_profiles",
+    actual = "{}",
+    visibility = ["//visibility:public"],
+)
+""".format(repository_ctx.attr.fallback_profiles or ":empty"))
+
+local_provisioning_profiles = repository_rule(
+    environ = ["HOME"],
+    local = True,
+    implementation = _local_provisioning_profiles,
+    attrs = dict(
+        fallback_profiles = attr.label(
+            allow_files = [".mobileprovision"],
+        ),
+    ),
+)
+
+def _local_provisioning_profile(ctx):
+    selected_profile_path = "{name}.mobileprovision".format(name = ctx.label.name)
```

Comment:
An optional `profile_name` would be useful. This could default to `name` if one isn't provided of course, but keeping the label constant but `select`-ing between different profile names is a use case that we currently have.

Reply:
One more rationale: target names can't have `:` in them, and Xcode managed profiles will be called something like `iOS Team Provisioning Profile: com.my.app`.

(Ignore for a moment why one would need Xcode managed profiles 😸)

Reply:
Running into this as well

Reply:
```suggestion
    selected_profile_path = "{name}.mobileprovision".format(name = ctx.attr.profile_name or ctx.attr.name)
```